libs/template/reader_test.go

```diff
@@ -0,0 +1,123 @@
+package template
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/flags"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuiltInReader(t *testing.T) {
+	exists := []string{
+		"default-python",
+		"default-sql",
+		"dbt-sql",
+	}
+
+	for _, name := range exists {
+		r := &builtinReader{name: name}
+		fs, err := r.FS(context.Background())
+		assert.NoError(t, err)
+		assert.NotNil(t, fs)
+
+		// Assert file content returned is accurate and every template has a welcome
+		// message defined.
+		fd, err := fs.Open("databricks_template_schema.json")
+		require.NoError(t, err)
+		b, err := io.ReadAll(fd)
+		require.NoError(t, err)
+		assert.Contains(t, string(b), "welcome_message")
+		assert.NoError(t, fd.Close())
+	}
+
+	r := &builtinReader{name: "doesnotexist"}
+	_, err := r.FS(context.Background())
+	assert.EqualError(t, err, "builtin template doesnotexist not found")
+
+	// Close should not error.
+	assert.NoError(t, r.Close())
+}
+
+func TestGitUrlReader(t *testing.T) {
+	ctx := context.Background()
+	cmd := cmdio.NewIO(ctx, flags.OutputJSON, strings.NewReader(""), os.Stdout, os.Stderr, "", "bundles")
+	ctx = cmdio.InContext(ctx, cmd)
+
+	var args []string
+	numCalls := 0
+	cloneFunc := func(ctx context.Context, url, reference, targetPath string) error {
+		numCalls++
+		args = []string{url, reference, targetPath}
+		err := os.MkdirAll(filepath.Join(targetPath, "a/b/c"), 0o755)
```

Comment:
> Join(targetPath, "a/b/c"

if you use join already, you should probably do Join(targetPath, a, b, c)

Reply:
done